app/controllers/uploads_controller.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class UploadsController < ApplicationController
+  include UploadValidations
+  before_action :validate_file_upload!, only: [:create]
+  before_action :set_file, only: %i[create]
+  before_action :clear_errors
+
+  def index
+  end
+
+  def create
+    redirect_to :rulesets
+  end
+
+  def set_file
+    session[:oas_file_name] = file_params[:oas_file].original_filename
+    session[:oas_file_body] = file_params[:oas_file].read
```

Comment:
This was written early as I reviewed the PR. The suggest of writing to disk here could be substituted to storing the file in the database via an ActiveRecord model.

Reply:
Cache store would also work well, and have automatic clean-up. The reason I didn't go for this off the bat though is that as it stands it would store the files on the application server itself, which isn't good, or in memory, which also isn't good.
If we can provision a redis backing for the cache store that would probably be a better approach.

Reply:
Using session is a bad idea. What if a user wants to compare version of the same file in two different browser tabs? What if we later add user authentication - will users find themselves logged out after 20 minutes?

Reply:
Comparing two versions of the same file in different browser tabs wouldn't be an issue, since both tabs would share the same session and same file, so if you hit two different spectral routes they would both be called on the same file giving comparable results.
If we add user authentication later we'll then have a user to associate the file with, we could easily switch to store it in their user record instead, so it will just be a couple of lines changed in the upload controller - but that's not in scope for the alpha anyway. There are issues around storing the files for long-term anyway: we would have to do capacity management to prevent storage filling up and not everyone might want people to know what they are working on. Keeping the files around only as long as we need them seems a good approach for the current goals.

Reply:
For clarification - The database is quite capable of storing JSON strings of a couple of megabytes each, and the database is a better place to store this data than in temporary files on the application server itself. In practical terms there is no real difference between storing this data in a string in the automatically-managed session table vs a manually defined model/table, other than we save on boilerplate code and the tying of the browser session to the table is automatic.
I do foresee that we might need to add a user account mechanism to this later, but the form of that has not yet been decided. Will will have a local user database, or will we use single sign-on across the wider suite of applications built by other teams? Will we want to store the files in the database associated to a user, or would be better off storing these files in an S3 bucket and keeping a reference to it in the database? Will we have clean-up jobs to clean up old/stale files and their associated database records? Will we have capacity monitoring in place?
Until these questions are answered and we have a design for the future iterations of the application, it makes most sense to me to store the JSON is a postgres-backed session, so we meet our immediate objectives without jumping the gun on where future sprints will lead us design-wise.

Reply:
I don't have a problem storing the data in a database. It is the use of session that I have a problem with. Use an ActiveRecord model to store that data and this will not only deal with this issue, but make other parts of the application work better. For example, Rails validation methods can be used and standard error handling can be used instead of the current bespoke method of capturing flash content. 

There are many options for tidying up stale input. A delayed job could do it and if there was a cron job to catch any items that slip through, we'd have a really robust solution.

Reply:
@RobNicholsGDS With respect, you've not read/addressed my points above properly - if we write model code we'll have to manually associate models with sessions and then have extra tables to clean up afterwards (since the session is needed anyway). Plus we will have to redo/refactor all of this again once the requirements on password auth etc. become clear.
As far as I'm concerned this solution is done, and it works. You've not really given a compelling argument yet as to why it is fundamentally flawed. I stand by the solution as sufficient to meet sprint objectives.
I'm not against doing it the way you suggest in the long run - but for the this sprint I'd appreciate it if we could just go with this and get on with the other things we need to do, and return to this once we have a proper design for the user journey and the requirements for possible long-term storage of the documents etc are properly understood.

Reply:
I am afraid I am unable to approve this PR as it is.

Reply:
If we are aiming for MVP which is what we should always be aiming for then using session or a cache is fine, I don't see why that specifically is a blocker. 

If you go down the route of storing the files in the DB then we need to know if there are any legal issues around storing open api documentation. Do we need to notify the user about this etc.

These issues may apply to storing files temporarily as well.